src/Parser/Paladin/Protection/Modules/Features/Consecration.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import Analyzer from 'Parser/Core/Analyzer';
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import { formatPercentage } from 'common/format';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+
+class Consecration extends Analyzer {
+  get uptime() {
+    console.info(this.selectedCombatant.getBuffUptime(SPELLS.CONSECRATION_BUFF.id));
+    return this.selectedCombatant.getBuffUptime(SPELLS.CONSECRATION_BUFF.id) / this.owner.fightDuration;
+  }
+
+  get uptimeSuggestionThresholds() {
+    return {
+      actual: this.uptime,
+      isLessThan: {
+        minor: 0.95,
+        average: 0.9,
+        major: .8,
+      },
+      style: 'percentage',
+    };
+  }
+
+  suggestions(when) {
+    when(this.uptimeSuggestionThresholds)
+        .addSuggestion((suggest, actual, recommended) => {
+          return suggest('Your Consecration uptime can be improved. Maintain it to reduce all incoming damage by a flat amount and refresh it during rotational downtime.')
+            .icon(SPELLS.CONSECRATION_CAST.icon)
+            .actual(`${formatPercentage(actual)}% Consecration uptime`)
+            .recommended(`>${formatPercentage(recommended)}% is recommended`);
+        });
+  }
+
+  statistic() {
+    return (
+      <StatisticBox
+        icon={<SpellIcon id={SPELLS.CONSECRATION_CAST.id} />}
+        value={`${formatPercentage(this.uptime)} %`}
```

Comment:
Why not use the same efficiency as you do for Shield of the Righteous?

Reply:
Same efficiency as in "count the hits taken like SotR"?
Thats kinda pointless, Pallys [new Mastery](http://bfa.wowhead.com/spell=76671/mastery-divine-bulwark) reduces all damage taken, not just phyiscal.
And there's no reason not to have 100% uptime with Consecration lasting 12sec on a 4.5sec (-haste) CD

Reply:
It was mostly using percent of dmg taken instead of percent uptime. Could have downtime while not actively tanking and running around, but I'm not sure how relevant that is. I agree showing numbers of hits doesn't make sense for conc though.